dali/test/python/conditionals/test_pipeline_conditionals.py

```diff
@@ -535,6 +535,81 @@ def conditional_pipe():
     compare_pipelines(*pipes, bs, iters)
 
 
+# Test if operators without positional inputs but with argument inputs are correctly handled
+# in the split/merge - so they are tracked in the local scope.
+
+
+def test_arg_inputs_scoped_tracking():
+    test_data_root = get_dali_extra_path()
+    caffe_db_folder = os.path.join(test_data_root, 'db', 'lmdb')
+
+    bs = 10
+    iters = 5
+    kwargs = {"batch_size": bs, "num_threads": 4, "device_id": 0, "seed": 42}
+
+    @experimental.pipeline_def(enable_conditionals=True, **kwargs)
+    def global_transform_pipe():
+        encoded, _ = fn.readers.caffe(path=caffe_db_folder)
+        decoded = fn.decoders.image(encoded, device="mixed")
+        pred = fn.random.coin_flip(dtype=types.DALIDataType.BOOL)
+        angle = fn.random.uniform(values=[10, 20, 30])
+        rotate_transform = fn.transforms.rotation(angle=angle)
+        if pred:
+            output = fn.warp_affine(decoded, matrix=rotate_transform)
+        else:
+            output = decoded
+        return output
+
+    @experimental.pipeline_def(enable_conditionals=True, **kwargs)
+    def scoped_transform_pipe():
+        encoded, _ = fn.readers.caffe(path=caffe_db_folder)
+        decoded = fn.decoders.image(encoded, device="mixed")
+        pred = fn.random.coin_flip(dtype=types.DALIDataType.BOOL)
+        angle = fn.random.uniform(values=[10, 20, 30])
```

Comment:
Can we have a test when you put the angle inside the if?

Reply:
There is a test that checks if pure generators are registered in the top scope (test_generators).